andrada12/assignment3/WordGridTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author andrada
+ */
+public class WordGridTest {
+    
+    public WordGridTest() {
+    }
+    
+    @BeforeClass
+    public static void setUpClass() {
+    }
+    
+    @AfterClass
+    public static void tearDownClass() {
+    }
+    
+    @Before
+    public void setUp() {
+    }
+    
+    @After
+    public void tearDown() {
+    }
+
+    /**
+     * Test of words method, of class WordGrid.
+     */
+    @Test
+    public void testWords() {
+        System.out.println("words");
+        char[][] chars = {{'a','a','r'}, {'t', 'c', 'd'}};
+        WordGrid instance = new WordGrid();
+        
+        ArrayList<String> result = instance.words(chars, new String[]{"cat", "cart", "card", "car"});
+        
+        /*Setting expected result */
+        ArrayList<String> expResult = new ArrayList<String>();
+        expResult.add("cat");
+        expResult.add("car");
+        expResult.add("card");
+        
+        assertEquals(expResult, result);
+    }
+    
+    @Test
+    public void testWords2() {
+        System.out.println("words");
+        char[][] chars = {{'s','k','e', 'r'}, {'t', 'y', 'd', 'p'}, {'z', 'n', 'p', 'l'}};
+        WordGrid instance = new WordGrid();
+        
+        ArrayList<String> result = instance.words(chars, new String[]{"sky", "key", "keen"});
+        
+        /*Setting expected result */
+        ArrayList<String> expResult = new ArrayList<String>();
+        expResult.add("sky");
+        expResult.add("key");
+        
+        
+        assertEquals(expResult, result);
+    }
+    
+    @Test
+    public void testWords3() {
+        System.out.println("words");
+        char[][] chars = {{'s','k','z', 'r'}, {'t', 'p', 'd', 'p'}, {'z', 'n', 'p', 'l'}};
+        WordGrid instance = new WordGrid();
+        
+        ArrayList<String> result = instance.words(chars, new String[]{"sky", "key", "keen"});
+       
+        /*Setting expected result */
+        ArrayList<String> expResult = new ArrayList<String>();
+        
+        assertEquals(expResult, result);
+    }
+    
+
+    
+
+    /**
+     * Test of dfs method, of class WordGrid.
+     */
+    @Test
+    public void testDfs() {
```

Comment:
These methods could be improved by making the name reflect what is being tested, instead of just numbering them. So for example:

testWordGridTraversesWholeTree() or testWordGridIgnoresVisited()